htdocs/user/class/api_users.class.php

```diff
@@ -430,6 +430,66 @@ public function put($id, $request_data = null)
 		}
 	}
 
+	/**
+	 * Update a users password
+	 *
+	 * @param   int     $id        User ID
+	 * @param	int		$send_password		Set this to 1 to have thenew password sent to the user
+	 * @param   int     $entity    Entity ID (valid only for superadmin in multicompany transverse mode)
+	 * @return  int                1 if password changed, 2 if password changed and sent
+	 *
+	 * @throws RestException 403 Not allowed
+	 * @throws RestException 404 User not found
+	 * @throws RestException 500 System error
+	 *
+	 * @url	GET {id}/setPassword
+	 */
+	public function setPassword($id, $send_password = 0, $entity = 1)
+	{
+		global $conf;
+
+		if (!DolibarrApiAccess::$user->hasRight('user', 'user', 'creer') && empty(DolibarrApiAccess::$user->admin)) {
+			throw new RestException(403, "setPassword on user not allowed");
+		}
+
+		$result = $this->useraccount->fetch($id);
+		if (!$result) {
+			throw new RestException(404, 'User not found, no password changed');
+		}
+
+		if (!DolibarrApi::_checkAccessToResource('user', $this->useraccount->id, 'user')) {
+			throw new RestException(403, 'User not allowed for login '.DolibarrApiAccess::$user->login);
+		}
+
+		if (isModEnabled('multicompany') && getDolGlobalString('MULTICOMPANY_TRANSVERSE_MODE') && !empty(DolibarrApiAccess::$user->admin) && empty(DolibarrApiAccess::$user->entity)) {
+			$entity = (!empty($entity) ? $entity : $conf->entity);
+		} else {
+			// When using API, action is done on entity of logged user because a user of entity X with permission to create user should not be able to
+			// hack the security by giving himself permissions on another entity.
+			$entity = (DolibarrApiAccess::$user->entity > 0 ? DolibarrApiAccess::$user->entity : $conf->entity);
+		}
+
+		// I do not like that the first parameter has to be an empty string :-()
+		// in user/card.php it is called like this setPassword($user, '')
+		// and my development server does not actually send email, so I can't really test this
+		$newpassword = $this->useraccount->setPassword(global $user, '');	// This will generate a new password
```

Comment:
`global $user;` needs to be on a separte line.

(Also, make sure that you are referring to the user you want to change the password for the user you want to because the code above seems to be checking access to the provides user "ressource", and here you seem to be updating the password of the user itself).

Reply:
thanks @mdeweerd, I think I forgot to push the commit I made locally this morning before going to work, latest commits gives no error other than it can not send email because I don't have email configured on my development dolibarr.